### PR TITLE
Dual boot - software upgrade support

### PR DESCRIPTION
**Pull request by ldts**
Please do not submit a Pull Request via github.  Our project makes use of
mailing lists for patch submission and review.  For more details please
see https://www.denx.de/wiki/U-Boot/Patches

### CODE REVIEW

File: arch/arm/dts/imx7ulpea-ucom-kit_v2.dts [hunk 1]
```diff
@@ -112,6 +111,12 @@
 			>;
 		};
 
+		pinctrl_system_reset: system-reset {
+			fsl,pins = <
+				IMX7ULP_PAD_PTC10__PTC10	0x20100
```
Comment (mike-scott) [inline arch/arm/dts/imx7ulpea-ucom-kit_v2.dts]: Let's set this to: 0x20000 as the driver won't set mux mode from this value (it's masked).

File: arch/arm/dts/imx7ulpea-ucom-kit_v2.dts [hunk 2]
```diff
@@ -69,7 +69,6 @@
 		};
 
 	};
-
```
Comment (mike-scott) [inline arch/arm/dts/imx7ulpea-ucom-kit_v2.dts]: Can we leave this line in place?

File: board/embeddedartists/mx7ulpea-ucom/mx7ulpea-ucom.c [hunk 3]
```diff
@@ -99,6 +99,18 @@ int board_init(void)
 	return 0;
 }
 
+/* will reset the M4 and the A7 */
+void board_m4_restart(void)
```
Comment (mike-scott) [inline board/embeddedartists/mx7ulpea-ucom/mx7ulpea-ucom.c:103]: Technically this should only apply to Aeler's HW (Atlas).  Not the EA 7ulp SoM.  We don't have a split out DTS file for that, but let me double-check if pulling PTC10 on the EA HW will cause problems.

**Review by mike-scott — CHANGES_REQUESTED** [review]

File: board/embeddedartists/mx7ulpea-ucom/mx7ulpea-ucom.c (hunk 3 above)
Comment (mike-scott) [reply board/embeddedartists/mx7ulpea-ucom/mx7ulpea-ucom.c:103]: This one is ok for now. Looks like PTC10 isn't being used by default on EA board (it's connected to J39 PIN47 labelled: TP_RST-GPIO).  U-boot pulling this shouldn't matter.
We may consider breaking out Aeler HW DTS at some point.

**Review by ricardosalveti — COMMENTED** [review]
Extra done in the commit message for '[FIO toup] imx: mx7ulpea-ucom: system reset function' (after signed-off-by). FIO tag for the same commit should probably be extras or similar, as this will not hit upstream.

**Comment by ricardosalveti** [comment]
Should be fine to have these changes done against the ea target, but we should probably have a card to make a board and dts specifically for aeler.

File: arch/arm/dts/imx7ulpea-ucom-kit_v2.dts (hunk 2 above)
Comment (ldts) [reply arch/arm/dts/imx7ulpea-ucom-kit_v2.dts]: I'll remove the dts changes. lets merge yours instead

**Comment by ldts** [comment]
> Extra done in the commit message for '[FIO toup] imx: mx7ulpea-ucom: system reset function' (after signed-off-by). FIO tag for the same commit should probably be extras or similar, as this will not hit upstream.

fixing

**Review by mike-scott — APPROVED** [review]
LGTM

**Pull request merged**